business/istio_config.go

```diff
@@ -1129,6 +1130,24 @@ func (in *IstioConfigService) IsGatewayAPI() bool {
 	return in.k8s.IsGatewayAPI()
 }
 
+func (in *IstioConfigService) IsAmbientEnabled() bool {
+
+	var cniNetwork map[string]any
+	istioConfigMap, err := in.k8s.GetConfigMap(config.Get().IstioNamespace, "istio-cni-config")
+	if err != nil {
+		log.Errorf("Error getting istio-cni-config configmap: %s ", err.Error())
+	} else {
+		err = yaml.Unmarshal([]byte(istioConfigMap.Data["cni_network_config"]), &cniNetwork)
+		if err != nil {
+			log.Errorf("Error reading istio-cni-config configmap: %s ", err.Error())
+		}
+		if cniNetwork["ambient_enabled"] == true {
```

Comment:
Should we have type check here, since I believe cniNetwork["ambient_enabled"] is a `any`?

Like:

```
enabledAmbient, ok := cniNetwork["ambient_enabled"].(bool)

if ok && enabledAmbient {
    return true
}

```

Reply:
Thanks for the catch!